src/python/pants/jvm/subsystems.py

```diff
@@ -35,3 +42,35 @@ def resolves(self) -> dict[str, str]:
     @property
     def default_resolve(self) -> str | None:
         return cast(str, self.options.default_resolve)
+
+    def resolves_for_target(self, target: Target) -> tuple[str, ...]:
```

Comment:
Note that this is checking that all resolves are recognized, which we did not do before.